File: C_Programming/Study/Q5_Assignment9.c
#include <stdio.h>

///////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  CountMult
// Output:          
// Description:     Assignment_9: Q.4: Write a program which accept Number from user & return differrence between 
//                                  summation of even digits and summation of odd digits.    
// Date:            30/05/2026
// Author:          Shital Ajit Nikam   
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////


int CountDiff(int iNo)
{
    int iDigit = 0;
    int iEvenCnt = 0;
    int iOddCnt = 0;
    int iSumDiff = 0;

   
    while(iNo != 0)
    {
        iDigit = iNo % 10;

        if(iDigit % 2 == 0)
        {
            iEvenCnt++;
        }

        if(iDigit % 2 != 0)
        {
            iOddCnt++;
        }

        iSumDiff = iEvenCnt - iOddCnt;

        iNo = iNo /10;
    }
    
    return  iSumDiff;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
        
    printf("Enter Number: \n");
    scanf("%d", &iValue);

    iRet = CountDiff(iValue);

    printf("Differrence between Summation of Digits is : %d",iRet);

    return 0;
}

/////////////////////////////////////////////////////////////////////////////////////
//Time Complexity : O(N)  where N = length of Number
/////////////////////////////////////////////////////////////////////////////////////
// Input : 2395
// Output : -15  (2-17)  
// Input : 1018
// Output : 6   (8-2)
// 
/////////////////////////////////////////////////////////////////////////////////////

File: Python_Programming/Competative/Q5_Assignment9.py
#//////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name :  ChkDivisible
#        
# Description:     Assignment_9: Q.5: Write a program which accept one number & check whether it is divisible 
#                                     by 3 & 5.
# Input  :         5
# Output :         125
#                                     
# Date:            24/05/2026
# Author:          Shital Ajit Nikam   
#
#//////////////////////////////////////////////////////////////////////////////////////////////////////////

def ChkDivisible(No1):
    if(No1 % 3 == 0 and No1 % 5 == 0):
        return True
    else:
        return False

    
def main():
    value1 = int(input("Enter the number: "))
        
    Ret = ChkDivisible(value1)   
    
    if(Ret == True):
        print(f"{value1} is divisible by 3 & 5") 
    else :
        print(f"{value1} is not divisible by 3 & 5")  
     
    
if __name__ == "__main__":
    main()